src/TypeScript.UnitTests/NodeJSLocator/Locators/GlobalPathNodeLocatorTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarLint.VisualStudio.TypeScript.NodeJSLocator.Locators;
+
+namespace SonarLint.VisualStudio.TypeScript.UnitTests.NodeJSLocator.Locators
+{
+    [TestClass]
+    public class GlobalPathNodeLocatorTests
+    {
+        private readonly string programFilesPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles), "nodejs\\node.exe");
+
+        [TestMethod]
+        public void Locate_FileNotFound_Null()
+        {
+            using var scope = CreateEnvironmentVariableScope();
+
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(It.IsAny<string>())).Returns(false);
+
+            var testSubject = CreateTestSubject(fileSystem.Object);
+            var result = testSubject.Locate();
+
+            result.Should().BeNull();
+        }
+
+        [TestMethod]
+        public void Locate_FileFoundInPath_FilePath()
+        {
+            const string filePath = "c:\\test\\node.exe";
+            using var scope = CreateEnvironmentVariableScope(Path.GetDirectoryName(filePath));
+
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(filePath)).Returns(true);
+
+            var testSubject = CreateTestSubject(fileSystem.Object);
+            var result = testSubject.Locate();
+
+            result.Should().Be(filePath);
+
+            fileSystem.VerifyAll();
+            fileSystem.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void Locate_FileNotFoundInPath_FileFoundInProgramFiles_FilePath()
+        {
+            using var scope = CreateEnvironmentVariableScope();
+
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(programFilesPath)).Returns(true);
```

Comment:
I have a problem testing the x86 fallback.

On my machine, I have both program files and x86. So I can mock the fileSystem to return false for one and true for the other.
However, on our build agent, only x86 exists -- so the two cases become identical and I can't do proper assertions.
So I'm only testing the first program files, not the second fallback.

Reply:
The only thing I can think of would be change one of the other environment variables like `ProgramFiles(x86)` to a known value to see if that has an effect on what the `GetEnvironmentFolder` returns.

However, if that does work then it could leave the machine in a bad state if the test is interrupted for some reason so it doesn't get to do the Dispose cleanup to reset the variable. This is only likely to happen if we are running the test in the IDE and stop the test before it completes.

Reply:
I definitely wouldn't want to risk sabotaging our personal machines, so it's probably not worth it.